### PR TITLE
Us116068/build service gets

### PR DESCRIPTION
**Pull request by josephsaffron**

### CODE REVIEW

File: src/components/d2l-teacher-course-creation-admin.js [hunk 1]
```diff
@@ -67,8 +67,12 @@ class TeacherCourseCreationAdmin extends BaseMixin(LitElement) {
 			.getAssociations()
 			.then(associationsArray => {
 				let i = 0;
-				associationsArray.map(association => association.RowId = i++);
-				this.associations = associationsArray;
+				if (associationsArray) {
+					associationsArray.map(association => association.RowId = i++);
+					this.associations = associationsArray;
+				} else {
+					this.associations = [];
```
Comment (BraydonKains) [inline src/components/d2l-teacher-course-creation-admin.js:74]:
```suggestion
					this.associations = Array();
```
The lit-analyzer likes to get upset about `[]` for some reason (although it doesn't look like it complained in CI)

File: package.json [hunk 2]
```diff
@@ -15,7 +15,7 @@
     "lint:eslint": "eslint . --ext .js,.html",
     "lint:lit": "lit-analyzer custom-teacher-course-creation.js demo test",
     "start": "es-dev-server --app-index demo/index.html --node-resolve --dedupe --open --watch",
-    "test": "npm run lint && npm run test:headless && npm run test:diff",
+    "test": "npm run test:headless && npm run test:diff && npm run lint",
```
Comment (TomMichielsD2L) [inline package.json:18]: lets remove diff for now

File: src/services/tccService.js [hunk 3]
```diff
@@ -1,18 +1,45 @@
...
+		let departments = [];
+		do {
+			const body = await this.getPagedDepartments(departmentInfo.Id, bookmark);
+			const pageInfo = body.PagingInfo;
+			departments = departments.concat(body.Items);
+			bookmark = pageInfo.Bookmark;
+			if (!pageInfo.HasMoreItems) {
+				break;
+			}
+		} while (bookmark);
```
Comment (TomMichielsD2L) [inline src/services/tccService.js:34]: nit: is this condition ever false?

Comment (josephsaffron) [reply src/services/tccService.js:34]: on the last page, the bookmark is null.

Comment (TomMichielsD2L) [reply src/services/tccService.js]: Perhaps we can just check has MoreItems?

**Pull request merged**